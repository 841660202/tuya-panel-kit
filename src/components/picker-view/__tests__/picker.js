import { requireNativeComponent, Picker } from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import PickerIos from '../index.ios';
import PickerAndroid from '../index.android';

describe('Picker Component', () => {
  it('basic render', () => {
    const ios = renderer.create(<PickerIos />).toJSON();
    expect(ios).toMatchSnapshot();

    const android = renderer.create(<PickerAndroid />).toJSON();
    expect(android).toMatchSnapshot();
  });
  it('should picker ios value change', () => {
    const fn = jest.fn();
    const component = renderer.create(
      <PickerIos>
        <PickerIos.Item label="Java" value="java" />
        <PickerIos.Item label="JavaScript" value="js" />
      </PickerIos>
    );
    const { instance } = component.root;
    instance.onValueChange('java');

    component.update(
      <PickerIos selectedValue='js' onValueChange={fn}>
        <PickerIos.Item label="Java" value="java" />
        <PickerIos.Item label="JavaScript" value="js" />
      </PickerIos>
    );
    expect(instance.state.selectedValue).toBe('js');

    instance.onValueChange('java');
    expect(instance.state.selectedValue).toBe('java');
    expect(instance.props.onValueChange).toHaveBeenCalled();

  });
  it('should picker android value change', () => {
    let mockRef = undefined;
    const fn = jest.fn();
    const component = renderer.create(
      <PickerAndroid>
        <PickerAndroid.Item label="Java" value="java" />
        <PickerAndroid.Item label="JavaScript" value="js" />
      </PickerAndroid>
    );
    const { instance } = component.root;
    instance._onChange({ nativeEvent: { newIndex: 0 } });

    component.update(
      <PickerAndroid selectedValue='js' onValueChange={fn} ref={ref => { mockRef = ref; }}>
        <PickerAndroid.Item label="Java" value="java" />
        <PickerAndroid.Item label="JavaScript" value="js" />
      </PickerAndroid>
    );
    expect(instance.state.selectedIndex).toBe(1);

    instance._picker = mockRef;
    instance._picker.setNativeProps = fn;
    instance._onChange({ nativeEvent: { newIndex: 0 } });
    expect(instance._picker.setNativeProps).toHaveBeenCalled();

  });
  it('should picker android item render', () => {
    const component = renderer.create(<PickerAndroid.Item label="Java" value="java" />).toJSON();
    expect(component).toMatchSnapshot();
  });
});

