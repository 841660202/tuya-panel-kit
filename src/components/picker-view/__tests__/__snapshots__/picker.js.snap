// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker Component basic render 1`] = `
<View
  style={Object {}}
>
  <RCTPicker
    items={Array []}
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    selectedIndex={0}
    style={
      Array [
        Object {
          "height": 216,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Picker Component basic render 2`] = `
<View
  style={
    Array [
      Object {
        "height": 216,
      },
      undefined,
    ]
  }
>
  <RCTPicker
    items={Array []}
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    selectedIndex={0}
    style={
      Array [
        Object {
          "height": 216,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Picker Component should picker android item render 1`] = `null`;
